ivy/__init__.py

```diff
@@ -1063,6 +1063,7 @@ def current_sub_backends():
 upcast_dtypes = False
 crosscast_dtypes = False
 cast_dtypes = lambda: downcast_dtypes and upcast_dtypes and crosscast_dtypes
+throw_error_for_dtypes = True
```

Comment:
let's get this working similar to any other global mode, with a setter, an unsetter and a stack. Also could you please rename it to like `dtype_error_mode` or something?